File: resources/expected/ros2/datatype-examples/strict/src/datatypes_system_cpp_pkg/src/base_code/consumer_consumer_runner.cpp
#include "datatypes_system_cpp_pkg/user_headers/consumer_consumer_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

consumer_consumer::consumer_consumer() : consumer_consumer_base()
{
    // Invoke initialize entry point
    initialize();

    PRINT_INFO("consumer_consumer infrastructure set up");
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto executor = rclcpp::executors::MultiThreadedExecutor();
    auto node = std::make_shared<consumer_consumer>();
    executor.add_node(node);
    executor.spin();
    rclcpp::shutdown();
    return 0;
}

File: resources/expected/ros2/python-datatype-examples/strict/src/datatypes_system_py_pkg/datatypes_system_py_pkg/base_code/consumer_consumer_runner.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rclpy.executors import MultiThreadedExecutor
from datatypes_system_py_pkg.user_code.consumer_consumer_src import consumer_consumer
#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================
def main(args=None):
    rclpy.init(args=args)
    node = consumer_consumer()
    executor = MultiThreadedExecutor()
    executor.add_node(node)
    executor.spin()
    rclpy.shutdown()

if __name__ == "__main__":
    main()
